lib/github.ml

```diff
@@ -91,8 +91,9 @@ type gh_link =
   | Pull_request of repository * int
   | Issue of repository * int
   | Commit of repository * commit_hash
+  | Compare of repository * compare_branches
 
-let gh_link_re = Re2.create_exn {|^(.*)/(.+)/(.+)/(commit|pull|issues)/([a-z0-9]+)/?$|}
+let gh_link_re = Re2.create_exn {|^(.*)/(.+)/(.+)/(commit|pull|issues|compare)/([a-zA-Z0-9/.\-_]+)/?$|}
```

Comment:
it will match potentially invalid urls like `commit/ZZZ-zz_`. Is it a problem?

Reply:
also the `/?` at the end no longer matches the trailing slash

Reply:
yep, that's what I was thinking but generally when it is an invalid link, it will 404 out and when using the `submatches` if I put one more branch for the compare the changes to the current code is very drastic.



> also the `/?` at the end no longer matches the trailing slash

unfortunately since I added the match to include `/` it no longer matches that one

Reply:
I don't think it's good to rely on the 404-ing. Maybe the redundant match should be removed from the regex, and you can parse/validate the last path fragment differently based on link type?

and ideally, here:

```ocaml
| Compare of repository * compare_branches
```

I would have it be like `repository * string * string` so that you've parsed/validated the two distinct branches (actually refs) at the time of type construction.

Reply:
okay yep, that sounds good! i'll add one more round of matches rather than changing the current one so that the previous code won't get changed too much

Reply:
@yasunariw actually, it is rather much harder to do verification of 2 distinct branches because we can compare across forked repos the way we set up our current `repo` information, it is much harder to get at the branch.  I think a good compromise is verifying that there are indeed two separate valid branch names (they might not exist in real life) and then running the compare API and letting GitHub tells us whether it's a good compare. also @Khady what do you think?